### PR TITLE
Added Maybe functor and getQueryParam

### PR DESCRIPTION
**Pull request by MarcoLPR**
Added Maybe functor, it allows the user to access object properties without worrying about null values

Added getQueryParam, a function that receives a string as an argument and it gets its value from the window.location

Re-worked objectValidate

### CODE REVIEW

File: src/Maybe.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+export class Maybe {
```
Comment (geoperez) [inline src/Maybe.ts]: You can give it a try to generics.

File: src/getQueryParam.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+export const getQueryParam = (param: string): string | number | null => {
```
Comment (geoperez) [inline src/getQueryParam.ts]: Just maybe

File: src/getQueryParam.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    if (window.location && window.location.href) {
```
Comment (geoperez) [inline src/getQueryParam.ts]: With this function, you will introduce a dependency of browser environment.

Comment (MarcoLPR) [reply src/getQueryParam.ts]: Even if I check if window exists?

**Comment by geoperez** [comment]
Yes

File: src/Maybe.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+    constructor(value: T) {
+        this.value = value;
+    }
+
+    public map(fn: Function): Maybe<T> {
+        return new Maybe(this.value !== null && this.value !== undefined ? fn(this.value) : null);
+    }
+
+    public getOrDefault(defaultValue: any): any {
```
Comment (geoperez) [inline src/Maybe.ts]: Default value should be T

Comment (MarcoLPR) [reply src/Maybe.ts]: Nope, I can return any defaultValue, for example:
If T is string or number, I can have defaultValue = null
No?

Comment (geoperez) [reply src/Maybe.ts]: Did you explore the possibility of nullable type as generic arg?

**Pull request merged**